experiments/faster_generation/utils.py

```diff
@@ -52,7 +53,7 @@ def run_model(args, processor_cls, model_cls, run_prediction_loop):
     tokenizer = processor_cls.from_pretrained(args.model)
 
     if args.max_gpu_memory is None:  # fails if it doesn't fit in a GPU
-        max_memory = {0: "100GiB", "cpu": "0GiB"}
+        max_memory = {0: "100GiB", "cpu": "50GiB"}
```

Comment:
n00b question: why do we need to modify `max_memory`?
For Llama2 70B ran on 8 A100 GPUs, I had to replace this line with:
`max_memory = None`
to get it to work

Reply:
haha for no good reason: long ago, when I wrote this script, my setup crashed if a) the model didn't fully fit on the GPU OR b) I didn't set a limit slightly below the device capacity -- e.g. 20GiB on a RTX3090 (24GB)

can and should be replaced

Reply:
Haha ... OK. I submitted a commit to keep max memory to None if it is not specified by the user